src/v2/services/mod.rs

```diff
@@ -38,4 +53,13 @@ pub trait Service {
     }
 
     fn parse_message(&self, message: Vec<u8>) -> Result<Self::Response, ServiceError>;
+
+    fn get_response(
+        &self,
+        ctx: &mut ReqRespCtx,
+        response_size: usize,
+    ) -> Result<Self::Response, ServiceError> {
+        let message = ctx.get_grpc_response(response_size)?;
+        self.parse_message(message)
+    }
```

Comment:
I think this is nice to be symmetrical to the dispatch, where it takes the ctx and handles retrieving the response through that and parsing to `Self::Response` type